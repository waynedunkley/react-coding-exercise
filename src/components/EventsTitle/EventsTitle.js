import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import Icon from '../Icon'
import titleIcon from '../../icons/vivid-angle-top-left.svg'

const EventsTitle = ({ classes, ready, total }) => (
  <h3 className={classes.title}>
    <Icon className={classes.titleIcon} symbol={titleIcon} />
    Results{ready && `: ${total} events found`}
  </h3>
)

EventsTitle.propTypes = {
  ready: PropTypes.bool,
  total: PropTypes.number
}

export default injectSheet({
  title: {
    paddingLeft: 20,
    position: 'relative'
  },
  titleIcon: {
    position: 'absolute',
    left: 0,
    top: 5
  }
})(EventsTitle)
