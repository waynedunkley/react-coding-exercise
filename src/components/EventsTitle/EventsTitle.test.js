import React from 'react'
import EventsTitle from './EventsTitle'
import { shallow } from 'enzyme'

describe('EventsTitle component', () => {
  it('renders without totals', () => {
    const wrapper = shallow(<EventsTitle ready={false} total={30} />)
    expect(wrapper).toMatchSnapshot('EventsTitle without values')
  })

  it('renders with totals', () => {
    const wrapper = shallow(<EventsTitle ready={true} total={30} />)
    expect(wrapper).toMatchSnapshot('EventsTitle with values')
  })
})