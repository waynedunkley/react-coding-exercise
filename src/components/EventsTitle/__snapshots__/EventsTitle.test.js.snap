// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTitle component renders with totals: EventsTitle with values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(EventsTitle)
    ready={true}
    total={30}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "title": "EventsTitle-title-0-1-1",
        "titleIcon": "EventsTitle-titleIcon-0-1-2",
      },
      "ready": true,
      "total": 30,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "title": "EventsTitle-title-0-1-1",
          "titleIcon": "EventsTitle-titleIcon-0-1-2",
        },
        "ready": true,
        "total": 30,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EventsTitle component renders without totals: EventsTitle without values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(EventsTitle)
    ready={false}
    total={30}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "title": "EventsTitle-title-0-1-1",
        "titleIcon": "EventsTitle-titleIcon-0-1-2",
      },
      "ready": false,
      "total": 30,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "title": "EventsTitle-title-0-1-1",
          "titleIcon": "EventsTitle-titleIcon-0-1-2",
        },
        "ready": false,
        "total": 30,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
